### PR TITLE
Normalize loss components

### PR DESCRIPTION
**Pull request by wiederm**
## Pull Request Summary

An easy way to normalize loss components is

$$
L = \sum_i \frac{L_i}{L_i^{\text{detached}}}
$$

which uses the componentwise loss and divides it by the detached value, scaling the loss to 1.

```
# Normalizing losses and combining them
combined_loss = loss_1 / loss_1.detach() + loss_2 / loss_2.detach()

# Backpropagation
combined_loss.backward()
```

### Key changes
Notable points that this PR has either accomplished or will accomplish.
 - [x] add an option to normalize the loss

### Associated Issue(s)
 - [x] issue #271 
## Pull Request Checklist
 - [x] Issue(s) raised/addressed and linked
 - [x] Includes appropriate unit test(s)
 - [x] Appropriate docstring(s) added/updated
 - [x] Appropriate .rst doc file(s) added/updated
 - [x] PR is ready for review

### CODE REVIEW

**Comment by wiederm** [comment]
@chrisiacovella , even if we don't want to follow up on this kind of normalization, I think it's still helpful to merge this PR. It adds the infrastructure for other loss scaling.
